tools/src/main/scala/scala/scalanative/build/LLVM.scala

```diff
@@ -0,0 +1,215 @@
+package scala.scalanative
+package build
+
+import java.nio.file.{Files, Path, Paths}
+
+import scala.collection.JavaConverters._
+import scala.util.Try
+import scala.sys.process._
+
+import IO.RichPath
+
+object LLVM {
+
+  /** Discover concrete binary path using command name and
+   *  a sequence of potential supported versions.
+   */
+  def discover(binaryName: String,
+               binaryVersions: Seq[(String, String)]): Path = {
+    val docSetup =
+      "http://www.scala-native.org/en/latest/user/setup.html"
+
+    val envName =
+      if (binaryName == "clang") "CLANG"
+      else if (binaryName == "clang++") "CLANGPP"
+      else binaryName
+
+    sys.env.get(s"${envName}_PATH") match {
+      case Some(path) => Paths.get(path)
+      case None => {
+        val binaryNames = binaryVersions.flatMap {
+          case (major, minor) =>
+            Seq(s"$binaryName$major$minor", s"$binaryName-$major.$minor")
+        } :+ binaryName
+
+        Process("which" +: binaryNames)
+          .lines_!(SilentLogger)
+          .map(Paths.get(_))
+          .headOption
+          .getOrElse {
+            throw new Exception(
```

Comment:
We should have a dedicated exception type for exceptions thrown with build api. Something along the lines of `sealed class BuildException(...) extends Exception(...)`.